install/install.go

```diff
@@ -660,11 +821,205 @@ func K8sInstaller(c *k8s.Client, o Options) error {
 
 	}
 	if verify && !o.Save {
-		checkPods(c, o)
+		checkPodsLegacy(c, o)
+	}
+	return nil
+}
+
+func actionConfigInit(ns string, settings *cli.EnvSettings) *action.Configuration {
+	actionConfig := new(action.Configuration)
+	if err := actionConfig.Init(settings.RESTClientGetter(), ns, os.Getenv("HELM_DRIVER"), log.Printf); err != nil {
+		fmt.Println("failed to initialize Helm configuration: " + err.Error())
+		return nil
 	}
+	return actionConfig
+}
+
+func writeHelmManifests(manifests string, filename string, printYAML []interface{}, kubearmorConfig *Operatorv1.KubeArmorConfig) error {
+	currDir, err := os.Getwd()
+	if err != nil {
+		log.Fatal(err)
+	}
+	printYAML = append(printYAML, kubearmorConfig)
+	if err != nil {
+		return err
+	}
+
+	f, err := os.Create(filepath.Clean(path.Join(currDir, "kubearmor.yaml")))
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := f.Close(); err != nil {
+			fmt.Printf("Error closing file: %s\n", err)
+		}
+	}()
+
+	for _, o := range printYAML {
+		if err := writeToYAML(f, o); err != nil {
+			return err
+		}
+	}
+
+	file, _ := os.OpenFile("kubearmor.yaml", os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0600)
+	// Write the string to the file
+	_, err = file.WriteString(manifests + "\n")
+	if err != nil {
+		return err
+	}
+
+	err = f.Sync()
+	if err != nil {
+		log.Fatal(err)
+	}
+	s3 := f.Name()
+	fmt.Println("🤩\tKubeArmor manifest file saved to \033[1m" + s3 + "\033[0m")
 	return nil
 }
 
+func getOperatorConfig(o Options) map[string]interface{} {
+	var operatorImagePullPolicy string = "Always"
+	var opeartorImageTag string = "latest"
+
+	if o.Local {
+		operatorImagePullPolicy = "IfNotPresent"
+	}
+	if o.Tag != "" {
+		opeartorImageTag = o.Tag
+	}
+	if o.ImageRegistry != "" {
+		o.OperatorImage = o.ImageRegistry + "/" + o.OperatorImage
+	}
+	return map[string]interface{}{
+		"kubearmorOperator": map[string]interface{}{
+			"image": map[string]interface{}{
+				"repository": o.OperatorImage,
+				"tag":        opeartorImageTag,
+			},
+			"imagePullPolicy": operatorImagePullPolicy,
+		},
+	}
+}
+
+// K8sInstaller using operator for karmor
+func K8sInstaller(c *k8s.Client, o Options) error {
+	var printYAML []interface{}
+	ns := o.Namespace
+	releaseName := "kubearmor-operator"
+	kubearmorConfig := getOperatorCR(o)
+	values := getOperatorConfig(o)
+	settings := cli.New()
+	settings.SetNamespace(ns)
+
+	actionConfig := actionConfigInit(ns, settings)
+
+	entry := &repo.Entry{
+		Name: "kubearmor",
+		URL:  "https://kubearmor.github.io/charts",
+	}
+
+	r, err := repo.NewChartRepository(entry, getter.All(settings))
+	if err != nil {
+		return fmt.Errorf("failed to create ChartRepository: %w", err)
+	}
+
+	r.CachePath = settings.RepositoryCache
+
+	if _, err := r.DownloadIndexFile(); err != nil {
+		return fmt.Errorf("failed to download repository index file: %w", err)
+	}
+
+	var repoFile repo.File
+	repoFile.Update(entry)
+	if err := repoFile.WriteFile(settings.RepositoryConfig, 0644); err != nil {
```

Comment:
Question: Do we know where this file gets written to? Does it get cleaned up later?

Reply:
This code piece basically performs `helm repo add kubearmor https://kubearmor.github.io/charts` and `helm repo update kubearmor`. It is stored in `$HOME/.config/helm/repositories.yaml` and it does not clean up after uninstall. Happens same with `helm uninstall`.